packages/ionio/src/Transaction.ts

```diff
@@ -280,17 +322,31 @@ export class Transaction implements TransactionInterface {
     }
 
     // check for blinding to be made
-    if (this.inputBlindingData.size > 0) {
-      if (this.outputBlindingPubKeys.size === 0)
+    if (this.inputBlinders.length > 0) {
+      const needsBlinding = this.pset.outputs.some((o: PsetOutput) =>
+        o.needsBlinding()
+      );
+      if (!needsBlinding)
         throw new Error(
           'if one confidential input is spent, at least one of the outputs must be blinded'
         );
 
-      await this.psbt.blindOutputsByIndex(
-        Psbt.ECCKeysGenerator(this.ecclib),
-        this.inputBlindingData,
-        this.outputBlindingPubKeys
+      const zkpValidator = new ZKPValidator();
+      const zkpGenerator = ZKPGenerator.fromOwnedInputs(this.inputBlinders);
+      const ownedInputs = await zkpGenerator.unblindInputs(this.pset);
```

Comment:
Question: this.inputBlinders is the list of unblinded prevouts? the naming is a bit misleading.
If yes, then you don't even need to call `await zkpGenerator.unblindInputs(this.pset);` because it returns exactly `this.inputBlinders`, you could simply reuse it below

Reply:
exactly.

woudl `inputUnblindData` be better name?

Good to know, we do not need that extra step. At the moment we force user to pass `unblindData` field indeed, not blinding keys.

Reply:
yeah `unblindeInputs` is definitely better